src/otx/core/data/caching/mem_cache_handler.py

```diff
@@ -188,10 +190,20 @@ def create(cls, mode: str, mem_size: int) -> MemCacheHandlerBase:
         else:
             world_size = 1
 
+        # Prevent CPU OOM issue
+        memory_info = psutil.virtual_memory()
+        available_cpu_mem = memory_info.available / GIB
+
         if world_size > 1:
             mem_size = mem_size // world_size
+            available_cpu_mem = available_cpu_mem // world_size
             logger.info(f"Since world_size={world_size} > 1, each worker a {mem_size} size memory pool.")
 
+        logger.info(f"Try to create a {mem_size} size memory pool.")
+        if available_cpu_mem < ((mem_size / GIB) + cls.CPU_MEM_LIMITS_GIB):
+            logger.info("No available CPU memory left, mem_size will be set to 0.")
```

Comment:
```suggestion
            logger.warning("No available CPU memory left, mem_size will be set to 0.")
```
How about changing it to `warning`?
And Current code sets mem cache size as 0 if there isn't sufficient memory, but how about adapting it according to memory size?
For example, if available memory is 16GiB, then set mem cache size as `16GiB / 30`.

Reply:
It's a good option. However, this memory checking is conducted before the construction of the DataLoader. So, after this checking, large CPU memory will be consumed by Dataset, DataLoader, and some functions (i.e. evaluation).

I just left 30GiB as a minimum bound since I saw the OTX using almost 30GiB when the large images were used at instance segmentation or rotated detection tasks. 

Since this logic was added to guarantee stability, more room could be safe :) 
Let's change the logic as you said after we get some experience.